k8s/plugin/src/resources/uo_client.rs

```diff
@@ -0,0 +1,165 @@
+    pub(crate) async fn new(
+        uri: Option<String>,
+        namespace: String,
+        kube_config_path: Option<std::path::PathBuf>,
+        timeout: humantime::Duration,
+    ) -> Option<Self> {
+        let (uri, client) = match uri {
+            None => {
+                let (uri, svc) = match kube_proxy::ConfigBuilder::default_upgrade()
+                    .with_kube_config(kube_config_path)
+                    .with_target_mod(|t| t.with_namespace(namespace))
+                    .build()
+                    .await
+                {
+                    Ok(result) => result,
+                    Err(error) => {
+                        println!("Failed to create upgrade operator client {:?}", error);
+                        return None;
+                    }
+                };
+                (uri, svc)
+            }
+            Some(uri) => {
+                let mut connector = hyper::client::HttpConnector::new();
+                connector.set_connect_timeout(Some(*timeout));
+                let client = hyper::Client::builder()
+                    .http2_keep_alive_timeout(*timeout)
+                    .http2_keep_alive_interval(*timeout / 2)
+                    .build(connector);
+                let service = tower::ServiceBuilder::new()
+                    .timeout(*timeout)
+                    .service(client);
+                (uri.parse::<Uri>().unwrap(), BoxService::new(service))
+            }
+        };
+
+        Some(UpgradeOperatorClient {
+            uri,
+            inner_client: client,
+            service_endpoint: (UPGRADE_OPERATOR_END_POINT.to_string())
+                .parse::<Uri>()
+                .unwrap(),
+        })
+    }
+
+    pub async fn apply_upgrade(&mut self) -> Result<Option<Vec<String>>, UpgradeClientError> {
+        let request_str = format!("{}{}", self.uri.clone(), self.service_endpoint,);
+        let request = http::Request::builder()
+            .method("PUT")
+            .uri(&request_str)
+            .body(hyper::body::Body::empty())
+            .unwrap();
+        let response = self
+            .inner_client
+            .ready()
+            .await?
+            .call(request)
+            .await
+            .unwrap();
+        if !response.status().is_success() {
+            let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
+            let text = String::from_utf8(body_bytes.to_vec()).unwrap_or_default();
+            return Err(UpgradeClientError::Response(text));
+        } else {
+            let body = hyper::body::aggregate(response.into_body()).await?;
+            let ug_response: UpgradeClientResponse = serde_json::from_reader(body.reader())?;
+            println!("Upgrade From :  {:?}", ug_response.current_version);
+            println!("Upgrade To :  {:?}", ug_response.target_version);
+            println!("Upgrade Status :  {:?}", ug_response.state);
```

Comment:
Can simply use Display trait:
```suggestion
            println!("Upgrade From: {}", ug_response.current_version);
            println!("Upgrade To: {}", ug_response.target_version);
            println!("Upgrade Status: {}", ug_response.state);
```